test/e2e/utils/utils.go

```diff
@@ -111,3 +116,41 @@ func LocateRepoFile(repopath string) (string, error) {
 
 	return "", errors.New("no file found, try to define PLUGINS_REPO_DIR pointing to the root of the repository")
 }
+
+func createKustomizationOverlay(namespace, base, overlay string) error {
+	relPath := ""
+	for range strings.Split(overlay[1:], "/") {
+		relPath = relPath + "../"
+	}
+	relPath = relPath + base[1:]
+
+	content := fmt.Sprintf("namespace: %s\nbases:\n  - %s", namespace, relPath)
+	return ioutil.WriteFile(overlay+"/kustomization.yaml", []byte(content), 0600)
+}
+
+// DeployFpgaWebhook deploys FPGA admission webhook to a framework-specific namespace.
+func DeployFpgaWebhook(f *framework.Framework, kustomizationPath string) {
+	if _, err := e2epod.WaitForPodsWithLabelRunningReady(f.ClientSet, "cert-manager",
+		labels.Set{"app.kubernetes.io/name": "cert-manager"}.AsSelector(), 1 /* one replica */, 10*time.Second); err != nil {
+		framework.Failf("unable to detect running cert-manager: %v", err)
+	}
+
+	tmpDir, err := ioutil.TempDir("", "fpgawebhooke2etest-"+f.Namespace.Name)
+	if err != nil {
+		framework.Failf("unable to create temp directory: %v", err)
+	}
+	defer os.RemoveAll(tmpDir)
+
+	err = createKustomizationOverlay(f.Namespace.Name, filepath.Dir(kustomizationPath), tmpDir)
```

Comment:
is `RunKubecltOrDie()` not able to set the namespace?

Reply:
No. I checked the source code of the E2E framework and it's not clear what `RunKubectlOrDie()` uses `namespace` for.

IIRC at some point the function didn't even accept `namespace` as a parameter.